storm-core/src/jvm/org/apache/storm/StormSubmitter.java

```diff
@@ -233,6 +236,23 @@ public static void submitTopologyAs(String name, Map stormConf, StormTopology to
                 if(topologyNameExists(conf, name, asUser)) {
                     throw new RuntimeException("Topology with name `" + name + "` already exists on cluster");
                 }
+
+                // Dependency uploading only makes sense for distributed mode
+                List<String> jarsBlobKeys;
+                List<String> artifactsBlobKeys;
+
+                DependencyUploader uploader = new DependencyUploader();
+                try {
+                    uploader.init();
+                    jarsBlobKeys = uploadDependencyJarsToBlobStore(uploader);
+                    artifactsBlobKeys = uploadDependencyArtifactsToBlobStore(uploader);
```

Comment:
Both `jarsBlobKeys` and `artifactsBlobKeys` should be removed from blob store when topology submission is failed.